common/authentication/postgresql/metadata.go

```diff
@@ -43,34 +48,61 @@ func (m *PostgresAuthMetadata) Reset() {
 	m.ConnectionMaxIdleTime = 0
 	m.MaxConns = 0
 	m.UseAzureAD = false
+	m.UseAWSIAM = false
 	m.QueryExecMode = ""
 }
 
+type InitWithMetadataOpts struct {
+	AzureADEnabled bool
+	AWSIAMEnabled  bool
+}
+
 // InitWithMetadata inits the object with metadata from the user.
 // Set azureADEnabled to true if the component can support authentication with Azure AD.
 // This is different from the "useAzureAD" property from the user, which is provided by the user and instructs the component to authenticate using Azure AD.
-func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, azureADEnabled bool) (err error) {
+func (m *PostgresAuthMetadata) InitWithMetadata(meta map[string]string, opts InitWithMetadataOpts) (err error) {
 	// Validate input
 	if m.ConnectionString == "" {
 		return errors.New("missing connection string")
 	}
-
-	// Populate the Azure environment if using Azure AD
-	if azureADEnabled && m.UseAzureAD {
+	switch {
+	case opts.AzureADEnabled && m.UseAzureAD:
 		m.azureEnv, err = azure.NewEnvironmentSettings(meta)
 		if err != nil {
 			return err
 		}
-	} else {
-		// Make sure this is false
+	case opts.AWSIAMEnabled && m.UseAWSIAM:
+		m.awsEnv, err = aws.NewEnvironmentSettings(meta)
+		if err != nil {
+			return err
+		}
+	default:
+		// Make sure these are false
 		m.UseAzureAD = false
+		m.UseAWSIAM = false
 	}
 
 	return nil
 }
 
+func (m *PostgresAuthMetadata) ValidateAwsIamFields() (string, string, string, error) {
+	awsRegion, _ := metadata.GetMetadataProperty(m.awsEnv.Metadata, "AWSRegion")
```

Comment:
Do not use `GetMetadataProperty` except in testing.

Instead, in the original metadata options struct you can define a pointer type if you want to be able to check whether an option was set or not. Not set means nil value in that case.

DecodeMetadata will marshal the metadata into the appropriate struct, and that's the only way you should access the metadata - via the metadata struct.

Reply:
I did it this way based on feedback prior in this PR to do it this way, and am doing it the same way that the Azure Environment Settings get picked up. See here in `bindings/azure/eventgrid/eventgrid.go` for the same setup that I am using for consistency.
```

func (a *AzureEventGrid) ensureInputBindingMetadata() error {
	a.metadata.azureTenantID, _ = metadata.GetMetadataProperty(a.metadata.properties, azauth.MetadataKeys["TenantID"]...)
	if a.metadata.azureTenantID == "" {
		return errors.New("metadata field 'azureTenantID' is empty in EventGrid binding")
	}
	a.metadata.azureClientID, _ = metadata.GetMetadataProperty(a.metadata.properties, azauth.MetadataKeys["ClientID"]...)
	if a.metadata.azureClientID == "" {
		return errors.New("metadata field 'azureClientID' is empty in EventGrid binding")
	}
	a.metadata.azureSubscriptionID, _ = metadata.GetMetadataProperty(a.metadata.properties, "subscriptionID", "azureSubscriptionID")
	if a.metadata.azureSubscriptionID == "" {
		return errors.New("metadata field 'azureSubscriptionID' is empty in EventGrid binding")
	}
```